src/magnet_uri.cpp

```diff
@@ -215,8 +215,12 @@ namespace libtorrent {
 			url = uri.substr(pos, find(uri, "&", pos) - pos);
 		}
 
-		string_view btih = url_has_argument(uri, "xt");
-		if (btih.empty())
+		string_view btih = unescape_string(url_has_argument(uri, "xt"), ec);
```

Comment:
you can't assign a temporary string to a ``string_view``. But, we don't really need to decode the string here, as far as I can tell. We just need to match the prefix of "urn%3abtih%3a" (and I suppose the upper case versions too).